GarbageTruck: optimize FBSAssist.AudioAssist.GetAudioWaveValue

mosirnik opened the pull request:
This method gets called 1-2 times a frame in an H scene.

mosirnik commented:
(Force-pushed a fix to a typo in a comment)

ManlyMarco reviewed (commented):
You can vastly simplify this transplier by using CodeMatcher, otherwise all looks swell.

mosirnik commented:
@ManlyMarco Thanks. I rewrote the code using CodeMatcher. Also I removed the use of `[ThreadStatic]` because the method being patched is inherently single-threaded (due to the use of a Unity method).
